core/rest-client/src/http_client.rs

```diff
@@ -73,6 +76,7 @@ impl Default for HttpClient {
 			timeout: None,
 			headers: Headers::new(),
 			authorization: None,
+			authenticated_connection: false,
 		}
```

Comment:
authenticated_connection flag to decide how to send the request. 
- false for default behavior of http_req lib: the tls connection is established if the server's root certificates belongs to the Mozilla's trusted root certificates. 
- true for new feature : the tls connection is established if the server's root certificates matches the supplied certificate